### PR TITLE
Fix documentation for gammapy.irf

### PR DESCRIPTION
**Pull request by cosimoNigro**
I have been using the `gammapy.irf` classes a lot lately and noticed the documentation was incomplete and 
contained references to classes that do not exist anymore.

All the `.rst` files containing the irf components description have been updated. 
I linked every IRF components to the corresponding description in the OGADF format and provided a snippet of code
opening and visualising the component of a run of the HESS DL3 DR1 
(I think more detailed functionalities are anyhow covered by the API and the tutorial notebooks).

Let me know if the new irf documentation structure sits well with you, if it is too little or to much detailed.
I am also not familiar with the PSF classes so if something has to be improved there let me know. 

Thank you,

Cosimo

### CODE REVIEW

File: docs/irf/psf.rst
Comment (adonath) [inline docs/irf/psf.rst:3]: Minor: maybe keep the explicit title? It's maybe easier to read...

File: docs/irf/bkg.rst
Comment (adonath) [inline docs/irf/bkg.rst:19]: See above...

File: docs/irf/bkg.rst
Comment (adonath) [inline docs/irf/bkg.rst]: I think you mean background here...

File: docs/irf/edisp.rst
Comment (adonath) [inline docs/irf/edisp.rst:21]: There is a bit more to say here: the `EDispKernel` is also integrated along the migration axis, to get the RMF matrix with a given "reco" energy binning....

File: docs/irf/psf.rst
Comment (adonath) [inline docs/irf/psf.rst:21]: Typo: I guess you mean PSF here...

File: docs/irf/psf.rst
Comment (adonath) [inline docs/irf/psf.rst:32]: For this we have introduced the convention to name it "rad", could be mentioned here...

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @cosimoNigro, I have a few first minor comments already. It would be good if @registerrier could do another review...

**Review by registerrier — CHANGES_REQUESTED** [review]
Thanks a lot @cosimoNigro !

This looks good. I have proposed inline a few changes to clarify some parts.

**Comment by cosimoNigro** [comment]
Hello @adonath and @registerrier,

I have implemented your comments.

As for the script for the 2D background generation from a list of OFF, shall I explicitly mention that it will be moved to the contributed notebooks or I just leave it like this and then we modify the link to it when we actually move it?

**Comment by cosimoNigro** [comment]
Another thing, the notation is a bit mixed: 
- in the IRF implementation description we use (E_true, E_reco), (p_true, p_reco);
- in the IRF theory we mix subscripted and non-subscripted (E, E_reco), (p, p_reco);
which one should we use?
I think if the reconstructed quantities are subscripted also the true quantities should be, for uniformity, so it should be
`A_{\rm eff}(E_{\rm true}, p_{\rm true})`
`PSF(P_{\rm recdo} | E_{\rm true}, p_{\rm true})`


what do you suggest? If we go for (E, E_reco), (p, p_reco) then the migration reads `\mu = E_{\rm reco} / E` and the rad reads `\delta p = p_{\rm true} - p`. But the integrals will have ugly differentials `d p_{\rm true}`.

Another option is priming the estimated quantities, so (E, E'), (p, p'). Though this does not relate and recall the axis naming convention.

Which one then?
- [ ] subscripted (E_true, E_reco), (p_true, p_reco);
- [ ] primed (E, E'), (p, p');
- [ ] mixed (E, E_reco), (p, p_reco) - as is now.

**Comment by cosimoNigro** [comment]
For the moment I reformatted everything to follow the third option: `(E, E_reco), (p, p_reco)`.
I removed all the `_{\rm true}` subscripts in the IRF implementation description.

**Comment by adonath** [comment]
Thanks @cosimoNigro! In the axis naming convention we use `energy` and `energy_true`, so maybe use `E` and `E_true` (or shorter like `E_t`) for the formulae? I'm not saying this is the clearest option, but the consistency with axis naming might be a plus. But I don't feel strongly there...

You can keep the reference to the background notebook, we will change it once the notebook actually is moved over....

**Comment by cosimoNigro** [comment]
Hi @adonath,

I have subscrtipted the true quantity `(E_{\rm true}, p_{\rm true})` and left unsubscripted the estimated one.
I added in the AXIS naming convention a reference to the notation used in theory.

I am good to merge if you do not have further requests.

**Comment by adonath** [comment]
@cosimoNigro Perfect, thanks a lot! Let's wait for the CI builds and then merge...

**Review by adonath — APPROVED** [review]
Thanks again @cosimoNigro, no further comments from my side...

**Comment by adonath** [comment]
Here are the rendered docs: https://docs.gammapy.org/dev/irf/index.html

**Pull request merged**
